products/llbuild-analyze/Sources/CriticalPathTool.swift

```diff
@@ -272,3 +291,119 @@ struct EncodableRuleResult: Encodable {
         try container.encode(deps, forKey: .dependencies)
     }
 }
+
+
+protocol GraphVizNode {
```

Comment:
Could we move all of that to a new file dedicated only for the graph viz output? We should do the same for the JSON serialization code, but that doesn't need to be in this PR.